### ISSUE TITLE
Join-SafeguardSessionCluster fails using updated versions of SPP and SPS

### ISSUE THREAD

**Issue by srehorn** [OneIdentity/safeguard-ps#311]
Repro: deploy marketplace images to Azure for SPP and SPS, apply patches to each via Support site downloads, run Join-SafeguardSessionCluster with suitable args. 

Yields unhelpful json parsing error when it attempts the meat of the API call for join, looks like this: 

`foo/.local/share/powershell/Modules/safeguard-ps/6.8.592/sessionjoin.psm1:540
Line |
 540 |          if (($_.ErrorDetails.Message | ConvertFrom-Json).error.detail …
     |                                         ~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'InputObject' because it is null.
`

Tried with SPS already set as cluster and primary node

get-module safeguard-ps yields `Script     6.8.592               safeguard-ps`
get-safeguardversion yields
`Major           : 6
Minor           : 8
Revision        : 0
Build           : 14830
BuildDate       : 12/09/2020 18:36:17
BuildVariant    : prod
BuildPlatform   : azure
SessionsVersion :
RsmsVersion     : 6.8.0.1207
Baseline        : False`

SPS patched to 6.7.2

**Comment by srehorn** [comment]
Additional note: when deploying via Azure marketplace, SPP incorporates the private subnet IP for its certificate, and this is rejected by SPS during the join process because SPS relies on a redirect to authenticate to SPP and will not accept the cert

**Comment by petrsnd** [comment]
It looks like the real error message is being masked by a failure in error handling.
```C#
    catch
    {
        if (($_.ErrorDetails.Message | ConvertFrom-Json).error.details.response.Code -eq 60657)
        {
            throw "This SPS cluster is already joined, check the output of Get-SafeguardSessionCluster."
        }
        throw
    }
```
I'll clean that up and find out what the real problem is.

**Comment by petrsnd** [comment]
@srehorn 
Issue is resolved in 6.8.627
